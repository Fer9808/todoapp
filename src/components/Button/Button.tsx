import React from "react";

import Text from '../Text/Text';
import {Â 
    ButtonContainer
} from './styles';
import * as ACTIONS from '../../types';


const Button = ({
    text,
    onPress,
    enableButton = true
} : ACTIONS.ButtonInterface) => {
    return (
        <ButtonContainer
            onPress={onPress}
            disabled={!enableButton}
        >
            <Text
                fontWeight={300}
                fontSize={'15px'}
                color={'white'}
            >
                { text }
            </Text>
        </ButtonContainer>
    );
}

export default Button;