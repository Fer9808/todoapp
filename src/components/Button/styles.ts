// @ts-ignore
import styled from 'styled-components/native';

interface PropsInterface {
    disabled: boolean
}

export const ButtonContainer = styled.TouchableOpacity`
    marginHorizontal: 20px;
    border-radius: 15px;
    background-color: #098606;
    alignItems: center;
    justifyContent: center;
    paddingVertical: 13px;
    marginVertical: 20px;
    opacity: ${(props: PropsInterface) => (props.disabled ? 0.6 : 1)}
`;