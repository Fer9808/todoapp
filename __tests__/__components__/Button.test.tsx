import * as React from 'react';
import { render } from '@testing-library/react-native';

import Button from '../../src/components/Button/Button';

describe('<Button />', () => {
    const  propsMockup = {
        text: 'Hola',
        onPress: jest.fn(),
        enableButton: true
    }

    const component = render(<Button { ...propsMockup } />)

    it('Render correctly Button Component', () => {
        expect(component).toMatchSnapshot();
    })
})
