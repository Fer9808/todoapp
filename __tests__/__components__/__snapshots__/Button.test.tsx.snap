// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Render correctly Button Component 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#098606",
      "borderRadius": 15,
      "justifyContent": "center",
      "marginHorizontal": 20,
      "marginVertical": 20,
      "opacity": 1,
      "paddingVertical": 13,
    }
  }
>
  <Text
    allowFontScaling={false}
    color="white"
    fontSize="15px"
    fontWeight={300}
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 15,
          "fontWeight": "300",
        },
      ]
    }
  >
    Hola
  </Text>
</View>
`;
